src/main/java/org/dependencytrack/upgrade/v4132/v4132Updater.java

```diff
@@ -0,0 +1,169 @@
+        }
+    }
+
+    private void createVulnerabilityCompositeSourceVulnIdIndex(final Connection connection) throws SQLException {
+        try (final Statement stmt = connection.createStatement()) {
+            LOGGER.info("Creating composite index on 'VULNERABILITY' for (SOURCE, VULNID)");
+            stmt.execute(/* language=SQL */ """
+                    CREATE INDEX IF NOT EXISTS "VULNERABILITY_COMPOSITE_SOURCE_VULNID_IDX"
+                      ON public."VULNERABILITY"("SOURCE", "VULNID")
+                    """);
+        }
+    }
+
+    private void createFindingAttributionCompositeIndex(final Connection connection) throws SQLException {
+        try (final Statement stmt = connection.createStatement()) {
+            LOGGER.info("Creating composite index on 'FINDINGATTRIBUTION' for (COMPONENT_ID, VULNERABILITY_ID, ATTRIBUTED_ON DESC)");
+            stmt.execute(/* language=SQL */ """
+                    CREATE INDEX IF NOT EXISTS "FINDINGATTRIBUTION_COMPOSITE_VULNERABILITY_ATTRIBUTED_IDX"
+                      ON public."FINDINGATTRIBUTION"(
+                        "COMPONENT_ID",
+                        "VULNERABILITY_ID",
+                        "ATTRIBUTED_ON" DESC
+                      )
+                    """);
+        }
+    }
+
+    private void createAnalysisCompositeIndices(final Connection connection) throws SQLException {
+        try (final Statement stmt = connection.createStatement()) {
+            LOGGER.info("Creating composite index on 'ANALYSIS' for (COMPONENT_ID, VULNERABILITY_ID, PROJECT_ID)");
+            stmt.execute(/* language=SQL */ """
+                    CREATE INDEX IF NOT EXISTS "ANALYSIS_COMPOSITE_VULNERABILITY_PROJECT_IDX"
+                      ON public."ANALYSIS"(
+                        "COMPONENT_ID",
+                        "VULNERABILITY_ID",
+                        "PROJECT_ID"
+                      )
+                    """);
+
+            LOGGER.info("Creating composite index on 'ANALYSIS' for (COMPONENT_ID, VULNERABILITY_ID, PROJECT_ID, SUPPRESSED)");
+            stmt.execute(/* language=SQL */ """
+                    CREATE INDEX IF NOT EXISTS "ANALYSIS_COMPOSITE_VULNERABILITY_PROJECT_SUPPRESSED_IDX"
+                      ON public."ANALYSIS"(
+                        "COMPONENT_ID",
+                        "VULNERABILITY_ID",
+                        "PROJECT_ID",
+                        "SUPPRESSED"
+                      )
+                    """);
+        }
+    }
+
+    private void createProjectActiveIndex(final Connection connection) throws SQLException {
+        try (final Statement stmt = connection.createStatement()) {
+            LOGGER.info("Creating partial index on 'PROJECT' for ACTIVE = true");
+            stmt.execute(/* language=SQL */ """
+                    CREATE INDEX IF NOT EXISTS "PROJECT_ACTIVE_IDX"
+                      ON public."PROJECT"("ID")
+                     WHERE "ACTIVE" = true
+                    """);
```

Comment:
Unfortunately, partial indexes are not supported by DataNucleus / JDO.

That is to say, users who deploy DT for the first time will not get this index, because this migration will never run for them.